### PR TITLE
RA-1283 Environmental Allergens are not working correctly

### PR DESCRIPTION
**Pull request by reubenvarghese1**
Now ,all of the allergens apart from food and drugs are shown under the "other" tab. This is as per the talk discussion here : https://talk.openmrs.org/t/two-bugs-about-allergies-and-appointments/9689/8 .The changes made are such that they support both Platform 1.9 and 2.x thus preserving backward compatibility.

https://issues.openmrs.org/browse/RA-1283

### CODE REVIEW

File: omod/src/main/webapp/pages/allergy.gsp [hunk 1]
```diff
@@ -60,11 +60,11 @@ ${ ui.includeFragment("allergyui", "removeAllergyDialog") }
             <div id="allergens">
 
                 <div id="types" class="button-group horizontal">
-                    <% allergenTypes.each { category -> %>
+                    <% allergenTypes.each { category ->  if (category.name()!="ENVIRONMENT") { %>
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/allergy.gsp]: What is the above if comparison doing?

Comment (reubenvarghese1) [inline omod/src/main/webapp/pages/allergy.gsp]: @dkayiwa This is done to comply with what Darius commented here :  https://github.com/openmrs/openmrs-module-allergyui/pull/26 . Basically this comparison ensures that a seperate tab for environment is not created in the UI although OpenMRS core supports it.

Comment (reubenvarghese1) [inline omod/src/main/webapp/pages/allergy.gsp]: @dkayiwa  Any update on this issue?

Comment (dkayiwa) [inline omod/src/main/webapp/pages/allergy.gsp]: So if my database has an allergy of type ENVIRONMENT, will it show up under the OTHER tab? If you are to say yes, have you actually tested and confirmed it?

Comment (reubenvarghese1) [inline omod/src/main/webapp/pages/allergy.gsp]: @dkayiwa Yes basically it's allergens show up under the "other" tab. I had attached the screenshots here already https://issues.openmrs.org/browse/RA-1283

File: omod/src/main/webapp/pages/allergy.gsp [hunk 2]
```diff
@@ -60,11 +60,11 @@ ${ ui.includeFragment("allergyui", "removeAllergyDialog") }
             <div id="allergens">
 
                 <div id="types" class="button-group horizontal">
-                    <% allergenTypes.each { category -> %>
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/allergy.gsp]: Doesn't category.name just work?

**Pull request merged**